scripts/lint-pr-checks.sh

```
@@ -0,0 +1,246 @@
+# Check pbxproj files for SWIFT_VERSION consistency (exclude third-party)
+for proj in $(find "$PROJECT_DIR" -name "project.pbxproj" \
+    -not -path "*/DerivedData/*" \
+    -not -path "*/.build/*" \
+    -not -path "*/build/*" \
+    -not -path "*/llama.cpp/*" \
+    -not -path "*/.venv/*" \
+    -not -path "*/node_modules/*"); do
+    versions=$(grep "SWIFT_VERSION" "$proj" | grep -oE '[0-9]+\.[0-9]+' | sort -u)
+    count=$(echo "$versions" | wc -l | tr -d ' ')
+    if [[ $count -gt 1 ]]; then
+        echo -e "   ${YELLOW}WARNING: $proj has multiple SWIFT_VERSION values: $(echo $versions | tr '\n' ' ')${NC}"
```

Shellcheck (0.11.0)

[info] Double quote to prevent globbing and word splitting.

(SC2086)